Ghostscript.NET.PDFA3Converter.Samples/ZUGFeRD/ZUGFeRDImporter.cs

```diff
@@ -0,0 +1,731 @@
+// CA 94129, USA, for further information.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using iText.Kernel.Pdf;
+using Microsoft.Extensions.Logging;
+
+/// <summary>
+/// ********************************************************************** Copyright 2018 Jochen Staerk Use is subject to license terms. Licensed under the
+/// Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at
+/// http://www.apache.org/licenses/LICENSE-2.0. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+/// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+/// and limitations under the License.
+/// </summary>
+namespace Ghostscript.NET.PDFA3Converter.Samples.ZUGFeRD
+{
+
+    /// <summary>
+    /// Mustangproject's ZUGFeRD implementation ZUGFeRD importer Licensed under the APLv2
+    /// 
+    /// @date 2014-07-07
+    /// @version 1.1.0
+    /// @author jstaerk
+    /// </summary>
+    public class ZUGFeRDImporter
+    {
+
+        /// <summary>
+        /// if metadata has been found
+        /// </summary>
+        //JAVA TO C# CONVERTER NOTE: Fields cannot have the same name as methods of the current type:
+        protected internal bool containsMeta_Conflict = false;
+        protected System.Xml.XmlDocument xmlDoc;
+        /// <summary>
+        /// map filenames of additional XML files to their contents
+        /// </summary>
+        private Dictionary<string, sbyte[]> additionalXMLs = new Dictionary<string, sbyte[]>();
+        /// <summary>
+        /// Raw XML form of the extracted data - may be directly obtained.
+        /// </summary>
+        private byte[] rawXML = null;
+        /// <summary>
+        /// XMP metadata
+        /// </summary>
+        private string xmpString = null; // XMP metadata
+        /// <summary>
+        /// parsed Document
+        /// </summary>
+        private XmlDocument document;
+
+        private readonly ILogger<ZUGFeRDImporter> _logger;
+
+
+        protected internal ZUGFeRDImporter()
+        {
+            //constructor for extending classes
+        }
+
+        public ZUGFeRDImporter(string pdfFilename)
+        {
+            using var loggerFactory = LoggerFactory.Create(builder =>
+        {
+            builder
+                .AddFilter("Microsoft", LogLevel.Warning)
+                .AddFilter("System", LogLevel.Warning)
+                .AddFilter("LoggingConsoleApp.Program", LogLevel.Debug);
+        });
+
+            if (!File.Exists(pdfFilename)) {
+                throw new Exception("File not found");
+            }
+
+            _logger = loggerFactory.CreateLogger<ZUGFeRDImporter>();
+            _logger.LogInformation("Example log message");
+            FromPDF(pdfFilename);
```

Comment:
in case of the failure of reading pdf data, we need to inform that to users, I think.
If not, they will get null reference exception in further operations with Importer.

Reply:
ZUGFeRDImporter removed